src/go/rpk/pkg/config/params.go

```diff
@@ -710,8 +710,30 @@ func (p *Params) Load(fs afero.Fs) (*Config, error) {
 		redpandaYaml: *DevDefault(),
 		rpkYaml:      defRpk,
 	}
-	c.rpkYaml.Profiles[0].KafkaAPI = c.redpandaYaml.Rpk.KafkaAPI
-	c.rpkYaml.Profiles[0].AdminAPI = c.redpandaYaml.Rpk.AdminAPI
+
+	// For Params, we clear the KafkaAPI and AdminAPI -- they are used to
+	// fill in rpk sections by default. These sections are *also* filled in
+	// in ensureBrokerAddrs, but only if we want a default cluster. At the
+	// end, if these two values are still nil AND there is no actual
+	// redpanda.yaml file with empty sections, we fill back in our old
+	// defaults.
```

Comment:
Could use some clarification on two points:

* at the end of what? command execution presumably
* why handle with a defer here if we are also dealing with the same values elsewhere?

Reply:
(resolved in convo in slack)